tt-server/server/models/location.py

```diff
@@ -1,7 +1,72 @@
+from pathlib import Path
+
+from geoalchemy2 import Geometry
+from geoalchemy2.shape import from_shape, to_shape
+from server.constants import UPLOAD_FOLDER
 from server.database import db
+from shapely import geometry
+from shapely.geometry import Point, Polygon
 
 
 class Location(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(64), nullable=False)
+    geometry = db.Column(Geometry(geometry_type="POLYGON", srid=4326))
     description = db.Column(db.Text, nullable=False)
+    last_updated = db.Column(db.DateTime, nullable=False)
+
+    @property
+    def images(self):
+        return Image.query.filter_by(location_id=self.id).all()
+
+    @classmethod
+    def from_points(cls, name, points, description, last_updated):
```

Comment:
If I'm understanding correctly, it seems like the location class will have a polygon defined by a set of points to designate a boundary. If the user is within the boundary, then they can add pictures? And how will the polygon points be defined?